extensions/message-tags.md

```diff
@@ -101,37 +147,44 @@ A new message command `TAGMSG` is defined for sending messages with tags but no
 This message MUST be delivered to targets in the same way as `PRIVMSG` and `NOTICE`
 messages. This means for example, honouring channel membership, modes,
 [`echo-message`](../extensions/echo-message-3.2.html),
-[`STATUSMSG`](https://tools.ietf.org/html/draft-hardy-irc-isupport-00#section-4.18)
+[`STATUSMSG`][statusmsg]
 prefixes, etc.
 
 Servers MAY apply moderation to this command using existing or newly specified modes or configuration.
 
 Servers MUST NOT deliver `TAGMSG` to clients that haven't negotiated the message tags capability.
 
-See [`PRIVMSG` in RFC2812](https://tools.ietf.org/html/rfc2812#section-3.3.1) for more details on replies and examples.
+See [`PRIVMSG` in RFC2812][privmsg] for more details on replies and examples.
 
 Clients that receive a `TAGMSG` command MUST NOT display them in the message history by default. Display guidelines are defined in the specifications of tags attached to the message.
 
-### Size limit
+## Rules for naming message tags
 
-The size limit for message tags is increased from 512 to 4607 bytes, including the leading `'@'` (0x40) and trailing space `' '` (0x20) characters. The size limit for the rest of the message is unchanged.
+Full tag names, including any vendor prefixes MUST be treated as an opaque identifier.
 
-This limit is separated between server-initiated and client-initiated tags. This prevents servers from overflowing the overall limit by adding tags to a client message sent within the allowed limit.
+There are two tag namespaces:
 
-In the following description, **tag data** describes the bytes between the leading and trailing tag separators mentioned above.
+### Vendor-Specific
 
-Clients MUST NOT send messages with tag data exceeding 4094 bytes, this includes tags with or without the client-only prefix.
+Names which contain a slash character (`/`) designate a vendor-specific tag namespace.
+These names are prefixed by a valid DNS domain name.
 
-Servers MUST NOT add tag data exceeding 510 bytes to messages.
+For example: `znc.in/server-time`.
 
-    <server_max>    (512)  :: '@' <tag_data  510> ' '
-    <client_max>   (4096)  :: '@' <tag_data 4094> ' '
-    <combined_max> (4607)  :: '@' <tag_data  510> ';' <tag_data 4094> ' '
+In cases where the prefix contains non-ASCII characters, punycode MUST be used,
+e.g. `xn--e1afmkfd.org/foo`.
 
-Servers MUST reply with the `ERR_INPUTTOOLONG` (`417`) error numeric if a client sends a message with more tag data than the allowed limit. Servers MUST NOT truncate tags and MUST always reject lines that exceed this limit.
+Vendor-Specific tags should be submitted to the IRCv3 working group for consideration.
 
-    417    ERR_INPUTTOOLONG
-          ":Input line was too long"
+### Standardized
+
+Names for which a corresponding document sits in the IRCv3 Extension Registry.
```

Comment:
this should link to the `[IRCv3 Tag Registry](https://ircv3.net/registry.html#tags)`

Reply:
Done